dbjobqueue: use background context when closing listener

lzap opened the pull request:
Hello,

when `Close()` is called in all cases it ends up with "Error unlistening for jobs in dequeue" because in the defer block then driver performs `Exec` call with the caller's context, it is already done.

This pull request includes:

- [x] adequate testing for the new functionality or fixed issue

lzap commented on cmd/osbuild-composer-dbjobqueue-tests/main_test.go:
```diff
@@ -64,6 +64,7 @@ func TestJobQueueInterface(t *testing.T) {
 
 	t.Run("maintenance-query-jobs-before", wrap(testJobsUptoByType))
 	t.Run("maintenance-delete-job-results", wrap(testDeleteJobResult))
+	t.Run("close", wrap(testClose))
```
This needs to be the last test in the suite, I am lazy yeah. Let me know how to implement this in a nicer way, perhaps to reopen the connection afterwards? :)

ondrejbudai replied:
Two ideas:

- put a comment before this subtest to indicate that it must be run last (lazier, gets the job done)
- make a completely separate test-case that spins up its own queue and then immediately close it (cleaner)

I'm fine with both. :)

lzap replied:
I was thinking exactly the same. Going for the cleanest solution.

I cannot figure out how to actually run those tests (I am still new to Go), so force-pushed blindly with fingers-crossed. If it fails, teach me please how to run them (this appear to be some kind of test suite, how do I trigger it)?

lzap commented:
Workflow still awaiting approval, some RPM builds failed as well.

ondrejbudai commented on cmd/osbuild-composer-dbjobqueue-tests/main_test.go:
```diff
@@ -144,3 +148,8 @@ func testDeleteJobResult(t *testing.T, q *dbjobqueue.DBJobQueue) {
 }
+
+func testClose(t *testing.T, q *dbjobqueue.DBJobQueue) {
+	err := q.Close()
```
`q.Close()` doesn't return any value, does it?

lzap replied:
Right, yeah. Hmmm, to test that I am going to check if a subsequent call returns an error. That's the only idea I have other than making something exported to check if the connection was closed.

ondrejbudai commented:
Hi @lzap, do you have any plans to get this finished? I'm currently in the process of cleaning up stale PRs. :) Thanks!

lzap commented:
Apologies, so I rebased this. I can't figure out how to test this, the error happens in the worker goroutine. On top of that, `Close` function is not part of the common interface. So I changed the patch so it only fixes the error, it's a cosmetic thing.

lzap commented on cmd/osbuild-composer-dbjobqueue-tests/main_test.go:
```diff
@@ -1,3 +1,4 @@
+//go:build integration
 // +build integration
```
My IDE insists on this, I did not fight it - I see few places where you still have this.

ondrejbudai replied:
Sure, `+build` will get removed when we switch to Go 1.18 completely (we cannot yet due to Fedora 35).

ondrejbudai commented:
Just an unrelated OSTree failure, merging.
